backend/service/bot/bot.go

```diff
@@ -0,0 +1,71 @@
+package bot
+
+import (
+	"fmt"
+
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/types/known/anypb"
+
+	botv1 "github.com/lyft/clutch/backend/api/config/service/bot/v1"
+	"github.com/lyft/clutch/backend/service"
+)
+
+const (
+	Name = "clutch.service.bot"
+)
+
+func New(cfg *anypb.Any, logger *zap.Logger, scope tally.Scope) (service.Service, error) {
+	config := &botv1.Config{}
+	if err := cfg.UnmarshalTo(config); err != nil {
+		return nil, err
+	}
+
+	t, err := getBotProvider(config)
+	if err != nil {
+		return nil, err
+	}
+
+	s := &svc{
+		logger:      logger,
+		scope:       scope,
+		botProvider: t,
+	}
+	return s, nil
+}
+
+func getBotProvider(cfg *botv1.Config) (botv1.Bot, error) {
+	switch cfg.BotProvider {
+	case botv1.Bot_SLACK:
+		return botv1.Bot_SLACK, nil
+	default:
+		return 0, fmt.Errorf("bot type '%v' not implemented", cfg.BotProvider)
+	}
+}
+
+type svc struct {
+	logger      *zap.Logger
+	scope       tally.Scope
+	botProvider botv1.Bot
+}
+
+type Service interface {
+	MatchCommand(command string) string
```

Comment:
any ideas for a better name? it sounds like you give it a command and it returns a command. but actually it returns the output of the command

Reply:
hmm, `getBotReply`/`getBotResponse` which would match what I have drafted in the module when we get back the match
```
match := m.bot.MatchSlackCommand(event.Text)
err := sendBotReply(m.slack, event.Channel, match)
```

Reply:
i guess it's fine as-is, maybe update the docstring to be more descriptive and give the return variable here a name